cdap-api/src/main/java/co/cask/cdap/api/lineage/Input.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.lineage;
+
+import co.cask.cdap.api.data.schema.Schema;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Represents input to the field level operation.
+ */
+public class Input {
+  private final String name;
+  private final Source source;
+
+  private Input(String name, @Nullable Source source) {
+    this.name = name;
+    this.source = source;
+  }
+
+  /**
+   * Return the Input name. The name could be complete path to the {@link Schema.Field}
+   * if it is created from it.
+   * @return the field
+   */
+  public String getField() {
```

Comment:
Shouldn't this method call `getName`??

Reply:
aah.. missed it changed the return type only.. fixed now.